pkg/csi/cinder/controllerserver.go

```diff
@@ -147,6 +167,38 @@ func (cs *controllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 	return getCreateVolumeResponse(vol, ignoreVolumeAZ, req.GetAccessibilityRequirements()), nil
 }
 
+func GetSnapshot(cloud openstack.IOpenStack, snapshotID string) (error, error) {
```

Comment:
if you're calling this func only from this go package, it's doesn't make sense to export the function. Moreover you can convert it to a `controllerServer` private method.